### PR TITLE
feat(ContactQuestioning): added contact status

### PR DESCRIPTION
**Pull request by roymaniv**

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx [hunk 1]
```diff
@@ -118,69 +78,19 @@ const ContactQuestioning: React.FC<Props> = ({ id, onSubmit }: Props): JSX.Eleme
                         allContactedInteractions.sort((firstInteractedContact, secondInteractedContact) =>
                         firstInteractedContact.phoneNumber ? firstInteractedContact.phoneNumber.localeCompare(secondInteractedContact.phoneNumber) : 0).map((interactedContact) => (
                                 <div key={interactedContact.id} className={classes.form}>
-                                    <Accordion expanded={interactedContact.expand} className={classes.accordion} style={{ borderRadius: '3vw'}}>
+                                    <Accordion className={classes.accordion} style={{ borderRadius: '3vw'}}>
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:81]: Why style?

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx [hunk 2]
```diff
@@ -118,69 +78,19 @@ const ContactQuestioning: React.FC<Props> = ({ id, onSubmit }: Props): JSX.Eleme
...
                                         <AccordionSummary
                                             test-id='contactLocation'
                                             expandIcon={<ExpandMore />}
-                                            onClick={() => {
-                                                updateInteractedContact(interactedContact, InteractedContactFields.EXPAND, !interactedContact.expand);
-                                            }}
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:127]: why did you delete it?

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx (hunk 1 above)
Comment (roymaniv) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:81]: Didn't change it, but doesn't work in class.

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx (hunk 2 above)
Comment (roymaniv) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:127]: not needed anymore

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 3]
```diff
@@ -45,6 +46,37 @@ ContactedPeopleRoute.get('/familyRelationships', (request: Request, response: Re
...
+            service: Service.SERVER,
+            severity: Severity.LOW,
+            workflow: 'Getting contact statuses',
+            step: 'got respond from the DB',
+            user: response.locals.user.id,
+            investigation: response.locals.epidemiologynumber
+        });
+        response.send(result)
+    }).catch(err => {
+        logger.error({
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:69]: send response with error code

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 4]
```diff
@@ -45,6 +46,37 @@ ContactedPeopleRoute.get('/familyRelationships', (request: Request, response: Re
...
+    logger.info({
+        service: Service.SERVER,
+        severity: Severity.LOW,
+        workflow: 'Getting contact statuses',
+        step: 'launching graphql API request',
+        user: response.locals.user.id,
+        investigation: response.locals.epidemiologynumber
+    });
+    graphqlRequest(GET_ALL_CONTACT_STATUSES, response.locals).then((result: any) => {
+        logger.info({
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:59]: Sometimes graphql return error on result, so you should check if the result is returned as exppceted.
In addition, I would change the route to return the array and not all the result, if you do so you need to check on the client the header application/json

**Review by sandymash — COMMENTED** [review]
maybe make status field a bit wider cause you cant really read it?
maybe even move phone a bit to the right?
image

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import PhoneDial from 'commons/PhoneDial/PhoneDial';
+import InteractedContact from 'models/InteractedContact';
+import InteractedContactFields from 'models/enums/InteractedContact';
+
+import useStyles from './ContactQuestioningStyles';
+
+const ContactQuestioningInfo: React.FC<Props> = (props: Props): JSX.Element => {
+    const classes = useStyles({});
+    
+    const { interactedContact, updateInteractedContact, contactStatuses } = props;
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo.tsx:19]: I think contactStatuses  need to be declared here as a state becuase it is only used here

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 6]
```diff
@@ -57,15 +57,26 @@ ContactedPeopleRoute.get('/contactStatuses', (request: Request, response: Respon
...
+                user: response.locals.user.id,
+                investigation: response.locals.epidemiologynumber
+            });
+            response.send(result?.data?.allContactStatuses?.nodes);
+        } else {
+            logger.error({
+                service: Service.SERVER,
+                severity: Severity.HIGH,
+                workflow: 'Getting contact statuses',
+                step: `got error from the graphql API ${JSON.stringify(result)}`,
```
Comment (sandymash) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:75]: i think error message is in result.errors

**Review by MayaSasson — DISMISSED** [review]

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts (hunk 6 above)
Comment (sandymash) [reply server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:75]: לפעמים הוא נופל עם error ולא מגיע לcatch אבל שם את השגיאה שם

Comment (sandymash) [reply server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:75]: image

Comment (sandymash) [reply server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:75]: https://sachee.medium.com/200-ok-error-handling-in-graphql-7ec869aec9bc

**Review by sandymash — DISMISSED** [review]

**Review by MayaSasson — APPROVED** [review]

**Review by EmilyLip — APPROVED** [review]
